### ISSUE TITLE
API Key issues with load_config() logic and config.json issues

### ISSUE THREAD

**Issue by jasonhatgithub** [xaitax/SploitScan#24]
Parsing API keys took a few stabs...

1) Config.json, despite creating in ~/.sploitscan/ will not be honoured in current logic if the default one provided is not removed, which isn't necessarily clear in the instructions/README
2) Config.json, kept complaining about "Error decoding JSON from the config file", and after adding some debugging to the code, python 3.12.3 on Ubuntu 24 complained about the extra spaces between colon : and start of variable in double quotes "Error decoding JSON from the config file /home/jasonh/SploitScan/sploitscan/config.json: Expecting property name enclosed in double quotes: line 2 column 1 (char 2)".  Removing the extra space helped.
3) Made some minor code changes to better handle errors during load_config, you may wish to consider these
```
def load_config():
    default_config = {"vulncheck_api_key": None, "openai_api_key": None}
    base_path = os.path.dirname(os.path.abspath(__file__))
    config_paths = [
        os.path.join(base_path, "config.json"),
        os.path.expanduser("~/.sploitscan/config.json"),
        os.path.expanduser("~/.config/sploitscan/config.json"),
        "/etc/sploitscan/config.json",
    ]

    for config_path in config_paths:
        if os.path.exists(config_path):
            try:
                print(f"Attempting to load config file from: {config_path}")  # Debugging statement
                with open(config_path, "r", encoding="utf-8") as file:
                    config = json.load(file)
                    print(f"Successfully loaded config file: {config_path}")  # Debugging statement
                    return config
            except json.JSONDecodeError as e:
                print(f"⚠️ Error decoding JSON from the config file {config_path}: {e}")
            except Exception as e:
                print(f"⚠️ Unexpected error reading config file {config_path}: {e}")
    
    print("⚠️ Config file not found in any checked locations, using default settings.")
    return default_config
```

**Comment by xaitax** [comment]
Thank you @jasonhatgithub - I believe this makes sense. Thank you!

**Comment by xaitax** [comment]
Fixed with [501be8e](https://github.com/xaitax/SploitScan/commit/501be8e8288ba50757e4e5e45bb73eb3919b4cc0)
